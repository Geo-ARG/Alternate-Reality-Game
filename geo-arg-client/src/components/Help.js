import React from 'react';
import { View, Text, Dimensions, Image, BackAndroid } from 'react-native';
import { Container, Header, Left, Right, Button, Title, Content } from 'native-base';
import Icon from 'react-native-vector-icons/MaterialIcons';
const { height, width } = Dimensions.get('window');

const styles = {
  header: {
    height: height * 0.1,
    backgroundColor: '#cc6600'
  },
  content: {
    height: height * 0.9
  },
  contentView: {
    position: 'absolute',
    alignSelf: 'center',
    backgroundColor: 'rgba(0,0,0,.7)',
    width: width,
    height: height
  },
  backgroundImg: {
    width: width,
    height: height
  },
  textView: {
    marginRight: 25,
    marginLeft: 25,
    marginTop: 20
  },
  textStyle1: {
    color: 'white', fontSize: 40, textAlign: 'center', fontWeight: 'bold', marginBottom: 20
  },
  textStyle2: {
    color: 'white', fontSize: 25, textAlign: 'justify', marginBottom: 10, fontWeight: 'bold'
  },
  textStyle3: {
    color: 'white', fontSize: 18, textAlign: 'justify'
  },
  textStyle4: {
    color: 'white', fontSize: 18, textAlign: 'justify', fontWeight: 'bold'
  },
  textStyle5: {
    color: 'white', fontSize: 24, textAlign: 'justify', marginTop: 10, marginBottom: 10, fontWeight: 'bold'
  }
}

export default class Help extends React.Component {
  constructor (props) {
    super(props)
  }

  render () {
    BackAndroid.addEventListener('hardwareBackPress', ()=> {
      this.props.navigator.pop()
      return true
    })
    return (
      <Container style={{backgroundColor: '#F5F5F5'}}>
        <Header style={styles.header}>
          <Left>
            <Button
              transparent
              onPress={() => this.props.navigator.pop()}
            >
              <Icon size={35} color={'white'} name='arrow-back' />
              <Title style={{fontSize: 25}}> Back</Title>
            </Button>
          </Left>
          <Right>
            <Title style={{fontSize: 25}}>Game Guides</Title>
          </Right>
        </Header>
        <Content style={styles.content}>
          <Image
            style={styles.backgroundImg}
            source={require('../assets/pokemonbg2.jpg')}
          />
          <View style={styles.contentView}>
            <View style={styles.textView}>
              <Text style={styles.textStyle1}>
                Geo-ARG Guides
              </Text>
              <Text style={styles.textStyle2}>
                Game objectives:
              </Text>
              <Text style={styles.textStyle3}>
                1. Find events near you and join them (you need to be near enough to be able to join the event)
              </Text>
              <Text style={styles.textStyle3}>
                2. Complete chain of quests that provided in each event.
              </Text>
              <Text style={styles.textStyle3}>
                3. Compete with your friends to get highest score and most achievements
              </Text>
              <Text style={styles.textStyle5}>
                Type of Quests:
              </Text>
              <Text style={styles.textStyle4}>
                Text:
              </Text>
              <Text style={styles.textStyle3}>
                Submit secret answer to complete the quest
              </Text>
              <Text style={styles.textStyle4}>
                Coordinate:
              </Text>
              <Text style={styles.textStyle3}>
                Go to a specific place to complete the quest
              </Text>
              <Text style={styles.textStyle4}>
                Photo:
              </Text>
              <Text style={styles.textStyle3}>
                Submit a photo and wait for admin to verify and complete the quest
              </Text>
            </View>
          </View>
        </Content>
      </Container>
    )
  }
}
