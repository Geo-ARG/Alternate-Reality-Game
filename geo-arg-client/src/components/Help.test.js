import renderer from 'react-test-renderer';
import React from 'react'
import Help from './help'

it('renders correctly', () => {
  const tree = renderer.create(
    <Help />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
