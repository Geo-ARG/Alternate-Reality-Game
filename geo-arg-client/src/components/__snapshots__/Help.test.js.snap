// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#cc6600",
          "borderBottomColor": "#a7a6ab",
          "borderBottomWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 133.4,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 15,
          "right": 0,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "top": 0,
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          }
        }
        virtual={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": "transparent",
              "borderColor": null,
              "borderRadius": 5,
              "borderWidth": null,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginLeft": -3,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 6,
              "paddingRight": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 35,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            î—„
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 25,
                "fontWeight": "600",
                "textAlign": "center",
              }
            }
            virtual={undefined}
          >
             Back
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 25,
              "fontWeight": "600",
              "textAlign": "center",
            }
          }
          virtual={undefined}
        >
          Game Guides
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    onScroll={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "height": 1200.6000000000001,
      }
    }
    virtual={undefined}
  >
    <View>
      <Image
        source={1}
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      />
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "rgba(0,0,0,.7)",
            "height": 1334,
            "position": "absolute",
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "marginLeft": 25,
              "marginRight": 25,
              "marginTop": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 40,
                "fontWeight": "bold",
                "marginBottom": 20,
                "textAlign": "center",
              }
            }
          >
            Geo-ARG Guides
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 25,
                "fontWeight": "bold",
                "marginBottom": 10,
                "textAlign": "justify",
              }
            }
          >
            Game objectives:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "textAlign": "justify",
              }
            }
          >
            1. Find events near you and join them (you need to be near enough to be able to join the event)
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "textAlign": "justify",
              }
            }
          >
            2. Complete chain of quests that provided in each event.
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "textAlign": "justify",
              }
            }
          >
            3. Compete with your friends to get highest score and most achievements
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 24,
                "fontWeight": "bold",
                "marginBottom": 10,
                "marginTop": 10,
                "textAlign": "justify",
              }
            }
          >
            Type of Quests:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "justify",
              }
            }
          >
            Text:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "textAlign": "justify",
              }
            }
          >
            Submit secret answer to complete the quest
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "justify",
              }
            }
          >
            Coordinate:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "textAlign": "justify",
              }
            }
          >
            Go to a specific place to complete the quest
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "justify",
              }
            }
          >
            Photo:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "textAlign": "justify",
              }
            }
          >
            Submit a photo and wait for admin to verify and complete the quest
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
          "margin": 20,
        }
      }
    >
      <View
        danger={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0,0,0,0.8)",
            "borderRadius": 5,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 50,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          }
        }
        success={false}
        virtual={undefined}
        warning={false}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          virtual={undefined}
        />
      </View>
    </View>
  </Modal>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.4)",
          "flex": 1,
          "justifyContent": "flex-end",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "elevation": 4,
            "height": NaN,
            "opacity": 1,
            "padding": 15,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#757575",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          virtual={undefined}
        />
        <RCTScrollView
          dataArray={Array []}
          dataSource={
            ListViewDataSource {
              "items": 0,
            }
          }
          enableEmptySections={true}
          renderRow={[Function]}
          renderScrollComponent={[Function]}
          style={
            Object {
              "marginLeft": -15,
              "marginRight": -15,
              "marginTop": 15,
            }
          }
          virtual={undefined}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
  </Modal>
</View>
`;
